crates/payload/basic/src/builder_stack.rs

```diff
@@ -0,0 +1,78 @@
+use crate::{
+    BuildArguments, BuildOutcome, MissingPayloadBehaviour, PayloadBuilder, PayloadBuilderError,
+    PayloadConfig,
+};
+
+/// A stack of payload builders that can be used in sequence.
+#[derive(Clone, Debug)]
+pub struct PayloadBuilderStack<B>(Vec<B>);
+
+impl<B> PayloadBuilderStack<B> {
+    /// Create a new payload builder stack.
+    pub fn new(builders: Vec<B>) -> Self {
+        Self(builders)
+    }
+
+    /// Push a builder onto the stack.
+    pub fn push(&mut self, builder: B) {
+        self.0.push(builder);
+    }
+}
+
+impl<B, Pool, Client> PayloadBuilder<Pool, Client> for PayloadBuilderStack<B>
+where
+    B: PayloadBuilder<Pool, Client>,
+{
+    type Attributes = B::Attributes;
+    type BuiltPayload = B::BuiltPayload;
+
+    fn try_build(
+        &self,
+        args: BuildArguments<Pool, Client, Self::Attributes, Self::BuiltPayload>,
+    ) -> Result<BuildOutcome<Self::BuiltPayload>, PayloadBuilderError> {
+        let mut args = Some(args);
```

Comment:
and these functions would then do a match on the `Either` and delegate the to A or B, we likely needs some mapper helpers on BuildArguments and Outcome